init.php

```diff
@@ -341,8 +341,10 @@ public function edit_user_profile_update( $user_id ) {
 			if ( ! function_exists( 'wp_handle_upload' ) )
 				require_once ABSPATH . 'wp-admin/includes/file.php';
 
-			// Delete old images if successful
-			$this->avatar_delete( $user_id );
+			// Delete old images if successful - This can cause issues with caching when enabled
+			if( apply_filters( 'basic_user_avatar_delete_avatar_before_upload', false ) ){
+				$this->avatar_delete( $this->user_id_being_edited );
+			}

```

Comment:
```suggestion
```suggestion
			/**
			 * Allow filtering whether to delete existing avatar for user before uploading new avatar.
			 *
			 * @since 1.0.7
			 *
			 * @param bool $delete_before_upload Wxhether to delete existing avatar for user before uploading new avatar.
			 */
			$delete_before_upload = apply_filters( 'basic_user_avatar_delete_avatar_before_upload', false );

			// Delete old images if successful - This can cause issues with caching when enabled
			if ( $delete_before_upload ) {
				$this->avatar_delete( $this->user_id_being_edited );
			}
```